.github/container/jax_nsys/python/jax_nsys/jax_nsys/analysis.py

```diff
@@ -193,13 +198,51 @@ def _get_message_size(
             "all-to-all",
             "collective-broadcast",
             "collective-permute-start",
+            "dynamic-slice",
+            "dynamic-update-slice",
             "reduce-scatter",
         }
     ), f"{instruction}: message size calculation for {comm_inst.opcode} has not yet been validated"
+
+    def _byte_size(inst) -> int:
+        size_bits = math.prod(
+            inst.shape.dimensions,
+            start=element_type_width(inst.shape.element_type),
+        )
+        size_bytes, rem = divmod(size_bits, 8)
+        assert rem == 0
+        return size_bytes
+
     if comm_inst.opcode == "collective-permute-start":
         # See https://openxla.org/xla/operation_semantics#collectivepermute, which
         # generates pair-wise send+recv between devices
         collective_size = 2
+    elif comm_inst.opcode in {"dynamic-slice", "dynamic-update-slice"}:
+        # Label host-device transfers orchestrated by dynamic[-update]-slice as single
+        # device collectives.
+        collective_size = 1
+        if comm_inst.opcode == "dynamic-update-slice":
+            # For dynamic-update-slice the second operand is the one being copied
+            _, src_inst = module_proto.find_instruction_by_id(comm_inst.operand_ids[1])
+            transfer_size = _byte_size(src_inst.proto())
+        else:
+            # For dynamic-slice the return type size is the transfer size
+            assert comm_inst.opcode == "dynamic-slice"
+            _, src_inst = module_proto.find_instruction_by_id(comm_inst.operand_ids[0])
+            transfer_size = _byte_size(comm_inst)
+        dest_on_host = _host_memory_space(comm_inst)
+        src_on_host = _host_memory_space(src_inst.proto())
+        assert src_on_host != dest_on_host, (
+            'dynamic[-update]-slice is only considered is only "communication" if it '
+            "represents a host-device transfer"
+        )
+        return (
+            transfer_size,
+            "device-to-host" if dest_on_host else "host-to-device",
```

Comment:
Can we exclude dynamic-update-slice and dynamic-slice if they don't have host memory space color 5 on neither source nor destination?

Reply:
There's an assert just above:
https://github.com/NVIDIA/JAX-Toolbox/blob/06ba7b3ec99abda2bbaff02da5d3959646dab945/.github/container/jax_nsys/python/jax_nsys/jax_nsys/analysis.py#L235-L238
that exactly one of source/destination is host memory (5), and it should never fail because of the logic here:
https://github.com/NVIDIA/JAX-Toolbox/blob/06ba7b3ec99abda2bbaff02da5d3959646dab945/.github/container/jax_nsys/python/jax_nsys/jax_nsys/protobuf.py#L82-L89
that only labels the op as being communication if the same condition is met.